shotover/src/server.rs

```diff
@@ -707,30 +707,29 @@ impl<C: CodecBuilder + 'static> Handler<C> {
                                 requests.extend(x);
                             }
                             debug!("Received requests from client {:?}", requests);
-                            if let Some(_close_reason) = self.send_receive_chain(local_addr, &out_tx, requests).await? {
-                                return Ok(())
+                            if let Some(close_reason) = self.process_requests(local_addr, &out_tx, requests).await? {
+                                return Ok(close_reason)
                             }
                         }
                         // Either we timed out the connection or the client disconnected, so terminate this connection
-                        None => return Ok(()),
+                        None => return Ok(CloseReason::ClientClosed),
                     }
                 },
             };
         }
 
-        Ok(())
+        Ok(CloseReason::ShotoverShutdown)
     }
 
-    async fn send_receive_chain(
+    async fn process_requests(
```

Comment:
A minor question, why do we rename this function again after it was renamed in the previous PR https://github.com/shotover/shotover-proxy/pull/1727/files#diff-a583948a329ee7447a9001d675ee60ab5789970c2b49cf275173357073199260R724 ?

Reply:
good catch, this was a bad merge, I'll fix it